### PR TITLE
amazon s3 e2e test for incorrect credential validation

### PR DESCRIPTION
**Pull request by itsmekumari**
amazon s3 e2e test for incorrect credential validation

### CODE REVIEW

**Review by itsankit-google — CHANGES_REQUESTED** [review]
Although the tests are failing but the `e2e-tests` build is succeeding, plz fix: 
image

**Comment by itsankit-google** [comment]
Although unrelated to this PR, but please fix the build workflow step here to use `head_sha` instead of `head_branch`: https://github.com/data-integrations/amazon-s3-plugins/blob/e6fe00e797045ec444154d83fb4c20aa01786811/.github/workflows/build.yml#L60

**Comment by itsmekumari** [comment]
> Although unrelated to this PR, but please fix the build workflow step here to use `head_sha` instead of `head_branch`:
> 
> https://github.com/data-integrations/amazon-s3-plugins/blob/e6fe00e797045ec444154d83fb4c20aa01786811/.github/workflows/build.yml#L60

Suggested Change done.

**Comment by itsmekumari** [comment]
> Although the tests are failing but the `e2e-tests` build is succeeding, plz fix: image

Screenshot 2023-11-22 at 19 11 33
Updated failsafe plugin goal to fix build issue

File: pom.xml [hunk 1]
```diff
@@ -284,7 +284,7 @@
   </dependencies>
 
   <build>
-    <testSourceDirectory>src/test/java</testSourceDirectory>
+    <testSourceDirectory>${testSourceLocation}</testSourceDirectory>
```
Comment (itsankit-google) [inline pom.xml:288]: Add `<testSourceLocation>${project.basedir}/src/test/java/</testSourceLocation>` in properties starting from line 76

Comment (itsmekumari) [reply pom.xml:288]: Added the Suggested change

File: pom.xml [hunk 2]
```diff
@@ -501,11 +504,37 @@
         </plugins>
       </build>
 
+
+      <dependencyManagement>
```
Comment (itsankit-google) [inline pom.xml:508]: add scope `test` to all dependencies.

Comment (itsmekumari) [reply pom.xml:508]: Added the scope test to all dependencies.

File: pom.xml [hunk 3]
```diff
@@ -501,23 +504,47 @@
...
+        <dependencies>
+          <dependency>
+            <groupId>com.google.guava</groupId>
+            <artifactId>guava</artifactId>
+            <version>31.1-jre</version>
+            <scope>test</scope>
```
Comment (itsankit-google) [inline pom.xml]: Do not need to add the scope here.

**Review by itsankit-google — COMMENTED** [review]
Changing the guava version is breaking the e2e test build:
image

File: pom.xml (hunk 3 above)
Comment (itsmekumari) [reply pom.xml]: Change done.

**Comment by itsmekumari** [comment]
> Changing the guava version is breaking the e2e test build: image

Resolved after removing the respective scope test.

File: pom.xml [hunk 4]
```diff
@@ -501,23 +504,44 @@
...
+
       <dependencies>
+        <dependency>
+          <groupId>com.google.guava</groupId>
+          <artifactId>guava</artifactId>
+        </dependency>
+        <dependency>
+          <groupId>org.slf4j</groupId>
+          <artifactId>slf4j-api</artifactId>
+          <version>2.0.7</version>
```
Comment (itsankit-google) [inline pom.xml]: scope test missing

File: pom.xml [hunk 5]
```diff
@@ -501,23 +504,44 @@
...
+            <artifactId>guava</artifactId>
+            <version>31.1-jre</version>
+          </dependency>
+        </dependencies>
+      </dependencyManagement>
+
       <dependencies>
+        <dependency>
+          <groupId>com.google.guava</groupId>
+          <artifactId>guava</artifactId>
```
Comment (itsankit-google) [inline pom.xml:526]: scope test missing

File: pom.xml (hunk 4 above)
Comment (itsmekumari) [reply pom.xml]: Getting compilation error in Build when scope test is added, Verified in local as well. Hence removed the tag.

File: pom.xml (hunk 5 above)
Comment (itsmekumari) [reply pom.xml:526]: Getting compilation error in Build when scope test is added, Verified in local as well. Hence removed the tag.

Comment (itsankit-google) [reply pom.xml:526]: Earlier it was added in dependency management not in the dependency.

Comment (itsmekumari) [reply pom.xml:526]: Earlier added in all dependency including dependency management, but later removed in these as compilation error in build occurred. Added now again to re-check and observed the same error. Hope will need to remove in these dependency. Please confirm so as to remove in L-522 and L-528

Comment (itsankit-google) [reply pom.xml:526]: We can just add both `sl4j-api` and `guava` in dependency management section and no need to add them as explicit dependency.

Comment (itsmekumari) [reply pom.xml:526]: Was able to add sl4j-api in dependency management and not guava as facing java.io.OutputStream com.google.common.io.ByteStreams.nullOutputStream() in build. Changes done accordingly.

**Review by itsankit-google — APPROVED** [review]
Please squash commits before merge.

**Comment by itsmekumari** [comment]
> Please squash commits before merge.

Squashed the commits

**Pull request merged**